browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,453 @@
+          base::UTF8ToUTF16(new_selectors_script), base::NullCallback(),
+          ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+  }
+
+  if (!enabled_1st_party_cf_filtering_) {
+    auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+    if (!frame_host)
+      return;
+    frame_host->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(*g_observing_script), base::NullCallback(),
+        ISOLATED_WORLD_ID_CHROME_INTERNAL);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::GetHiddenClassIdSelectorsOnUI(
+    content::GlobalFrameRoutingId frame_id,
+    const GURL& url,
+    std::unique_ptr<base::ListValue> selectors) {
+  if (!selectors || IsVettedSearchEngine(url.host())) {
+    return;
+  }
+  for (size_t i = 0; i < selectors->GetSize(); i++) {
+    base::ListValue* selectors_list = nullptr;
+    if (!selectors->GetList()[i].GetAsList(&selectors_list) ||
+        selectors_list->GetSize() == 0) {
+      continue;
+    }
+    std::string json_selectors;
+    if (!base::JSONWriter::Write(*selectors_list, &json_selectors) ||
+        json_selectors.empty()) {
+      json_selectors = "[]";
+    }
+    // Building a script for stylesheet modifications
+    std::string new_selectors_script =
+        base::StringPrintf(kSelectorsInjectScript, json_selectors.c_str());
+    if (selectors_list->GetSize() != 0) {
+      auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+      if (!frame_host)
+        return;
+      frame_host->ExecuteJavaScriptInIsolatedWorld(
+          base::UTF8ToUTF16(new_selectors_script), base::NullCallback(),
+          ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+  }
+
+  if (!enabled_1st_party_cf_filtering_) {
+    auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+    if (!frame_host)
+      return;
+    frame_host->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(*g_observing_script), base::NullCallback(),
+        ISOLATED_WORLD_ID_CHROME_INTERNAL);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::ProcessURL(
+    content::RenderFrameHost* render_frame_host,
+    const GURL& url,
+    const bool do_non_scriptlets) {
+  content::CosmeticFiltersCommunicationImpl::CreateInstance(render_frame_host,
+                                                            this);
+  if (!render_frame_host || !ShouldDoCosmeticFiltering(web_contents(), url)) {
+    return;
+  }
+  g_brave_browser_process->ad_block_service()
+      ->GetTaskRunner()
+      ->PostTaskAndReplyWithResult(
+          FROM_HERE,
+          base::BindOnce(GetUrlCosmeticResourcesOnTaskRunner, url.spec()),
+          base::BindOnce(
+              &BraveCosmeticResourcesTabHelper::GetUrlCosmeticResourcesOnUI,
+              AsWeakPtr(),
+              content::GlobalFrameRoutingId(
+                  render_frame_host->GetProcess()->GetID(),
+                  render_frame_host->GetRoutingID()),
+              url.spec(), do_non_scriptlets));
+}
+
+void BraveCosmeticResourcesTabHelper::DidFinishNavigation(
```

Comment:
from comments:
```
// Called when a navigation finished in the WebContents. This happens when a
// navigation is committed, aborted or replaced by a new one.
```
we probably shouldn't  do any processing if it was aborted or replaced.

also
```
  // If this is called because the navigation committed, then the document load
  // will still be ongoing in the RenderFrameHost returned by
  // |navigation_handle|.
```
mb we should used `DidStopLoading` instead

Reply:
yeah you are right about executing if committed only, I missed that. It wouldn't be anything bad if we do with not committed I think but let's put the check. That's fine that we execute scripts while document is still loading, we do several passes for purpose. I don't think we should change to `DidStopLoading`

Reply:
I've added the check